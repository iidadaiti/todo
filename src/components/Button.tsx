import '@spectrum-css/button'
import React, { Children, ComponentProps, FC, PropsWithChildren } from 'react'
import { mergeClassNames } from '../utils/classNames'

export interface ButtonProps
  extends Omit<ComponentProps<'button'>, 'children'> {
  size?: 's' | 'm' | 'l' | 'xl'
}

export const Button: FC<PropsWithChildren<ButtonProps>> = ({
  size = 'm',
  className,
  children,
  ...props
}) => (
  <button
    className={mergeClassNames(
      'spectrum-Button',
      'spectrum-Button--cta',
      `spectrum-Button--size${size.toUpperCase()}`,
      className
    )}
    {...props}
  >
    {Children.map(children, (child) =>
      typeof child === 'string' || typeof child === 'number' ? (
        <span className="spectrum-Button-label">{child}</span>
      ) : (
        child
      )
    )}
  </button>
)
