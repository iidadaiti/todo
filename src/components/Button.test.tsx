import { render } from '@testing-library/react'
import React from 'react'
import { Button } from './Button'

test('Button snapshot', () => {
  const result = render(<Button />)
  expect(result.asFragment()).toMatchSnapshot()
})
