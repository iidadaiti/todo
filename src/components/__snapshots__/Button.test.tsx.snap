// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<DocumentFragment>
  <button
    class="spectrum-Button spectrum-Button--cta spectrum-Button--sizeM"
  />
</DocumentFragment>
`;
